stock_picker.rb

```diff
@@ -0,0 +1,22 @@
+
+def stock_picker(prices_array)
+    highest_interval = 0
+    buy_day = 0
+    sell_day = -1
+    prices_array.each_with_index do |price, index|
+        break if index == prices_array.size - 1
+            prices_after_index = prices_array[index+1..-1]
+            max_value = prices_after_index.max
+            new_interval = max_value - price
+            if new_interval > highest_interval
+                highest_interval = new_interval
+                buy_day = index
+                sell_day = prices_after_index.index(max_value) + index + 1
+            end
+        end
+    [buy_day, sell_day]
+end
```

Comment:
what you are doing here
* you are iterating over the array - taking the current buying_price, and index
* you are checking for the current buying_price what's the maximal profit for it
* if it's bigger than the one we found so far - we replace the highest profit and highest profit interval

so:
* it's not `|price, index|` it's |buying_price, buy_day_index|
* it's not `max_value` it's `max_selling_price_for_buy_day`
* it's not `new_interval` (very confusing name for profit - it's not interval) - it's `max_profit_for_buy_day`
```suggestion
  prices_array.each_with_index do |buying_price, buy_day_index|
    break if buy_day_index == prices_array.size - 1

    max_selling_price_for_buy_day = prices_array[buy_day_index + 1..-1].max
    max_profit_for_buy_day = max_selling_price_for_buy_day - buying_price
    next unless max_profit_for_buy_day > highest_profit

    highest_profit = max_profit_for_buy_day
    highest_profit_interval = [buy_day_index, prices_array.index(max_selling_price_for_buy_day)]
  end
  highest_profit_interval
```

Reply:
if I use that code then the array `[6, 8, 17, 15, 7, 6, 3, 9, 10, 15, 1]` would return `[6, 3]` (so selling before buying, doesn't make sense) since `prices_array.index(max_selling_price_for_buy_day)` will return the first index of that value in the original array - so will fail with duplicates. This is why I also needed to keep the logic that finds the max in the array starting from the buy_day_index and add back the rest of the prices_array to get the index of the "real" max.
I used your suggestions for renaming but kept that logic.